pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -245,12 +247,26 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
     def expval(self, observable, shot_range=None, bin_size=None):
         if observable.name in [
             "Projector",
-            "Hamiltonian",
-            "SparseHamiltonian",
         ]:
             self.syncD2H()
             return super().expval(observable, shot_range=shot_range, bin_size=bin_size)
 
+        if observable.name in ["SparseHamiltonian"] and len(self.wires) >= 13:
+            CSR_SparseHamiltonian = observable.sparse_matrix().tocsr()
+            return self._kokkos_state.ExpectationValue(
+                CSR_SparseHamiltonian.data,
+                CSR_SparseHamiltonian.indices,
+                CSR_SparseHamiltonian.indptr,
+            )
+
+        if observable.name in ["Hamiltonian"] or (
+            observable.name in ["SparseHamiltonian"] and len(self.wires) < 13
+        ):
+            device_wires = self.map_wires(observable.wires)
+            return self._kokkos_state.ExpectationValue(
+                device_wires, qml.matrix(observable).ravel(order="C")
+            )
+
         if self.shots is not None:
             # estimate the expectation value
             samples = self.sample(observable, shot_range=shot_range, bin_size=bin_size)
```

Comment:
We should add explicit errors for attempting to use expvalH with adjoint, as this isn't currently supported.

Reply:
Good idea, added